Sources/Post.php

```diff
@@ -559,25 +569,28 @@ function Post($post_errors = array())
 			fatal_lang_error('no_message', false);
 		$row = $smcFunc['db_fetch_assoc']($request);
 
+		$context['do_not_select_character'] = true;
+		$context['posting_as_id'] = $row['id_character'];
+
 		$attachment_stuff = array($row);
 		while ($row2 = $smcFunc['db_fetch_assoc']($request))
 			$attachment_stuff[] = $row2;
 		$smcFunc['db_free_result']($request);
 
-		if ($row['id_member'] == $user_info['id'] && !allowedTo('modify_any'))
+		if ($row['id_member'] == $user_info['id'] && !charactersAllowedTo('modify_any'))
```

Comment:
is `charactersAllowedTo` different than `areCharactersAllowedTo`?

Reply:
Yes. allowedTo and charactersAllowedTo are soft checks, they return true/false. isAllowedTo and areCharactersAllowedTo are hard checks - failing them hits an error page directly (or login page for guests)